pcaplua.c

```diff
@@ -82,6 +104,41 @@ static int next (lua_State *L) {
 	return 3;
 }
 
+/** gets one packet
+ * @memberof pcap
+ * @return packet data, timestamp, offwire length
+ */
+static int next_ex (lua_State *L) {
+	l_pcap *p = check_pcap(L,1);
+	struct pcap_pkthdr *ph;
+	const u_char *d;
+	int res = pcap_next_ex (p->pcap, &ph, &d);
+	if (0 == res) {
```

Comment:
these conditions seem to be better as a `switch (res)`

Reply:
OK